src/ray/raylet/virtual_cluster_manager.cc

```diff
@@ -38,6 +38,13 @@ bool VirtualClusterManager::UpdateVirtualCluster(
       return false;
     }
 
+    // The local node is removed from a (indivisible) virtual cluster, we have to clean up
```

Comment:
Now the `virtual_cluster_data` is only removed from virtual_clusters_ when `virtual_cluster_data.is_removed()`, but the raylet may not receive the removed publish event, so that the virtual_cluster_data(removed from gcs) will never be erased at raylet side.